common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -4285,6 +4285,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
             "Bloom filter should be of at max certain size to be effective"),
     TEZ_BLOOM_FILTER_FACTOR("hive.tez.bloom.filter.factor", (float) 1.0,
             "Bloom filter should be a multiple of this factor with nDV"),
+    TEZ_BLOOM_FILTER_MERGE_THREADS("hive.tez.bloom.filter.merge.threads", 8,
```

Comment:
Any chance we can get rid of the extra configuration?
I really hope one day we will starting removing configuration instead of adding new :)

Reply:
agree, you'll see that the new version of PR will default to 0 which causes automatic thread number calculation

but I would leave this option as an "expert setting"...e.g.: explicitly setting a value for testing purposes, or setting -1 for sanity check (it will fail with negative values, so you can double-check if bloom filter merge really happens in your query)